### PR TITLE
#151297880 Embed Social Content

### PR DESCRIPTION
**Pull request by mazma1**
#### What does this PR do?
* Embed social content such as product reviews, tweets and posts from social media accounts

#### Description of Task to be completed
* Embed product reviews, tweets, and Facebook posts

#### How should this be manually tested?
* By submitting reviews for available products

#### What are the relevant pivotal tracker stories?
* Embed social content: #151297880

screen shot 2017-10-12 at 10 41 18 am

### CODE REVIEW

File: imports/plugins/included/product-detail-simple/client/templates/productReview.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Template.submitReview.events({
+  "click .stars": () => {
+    const rating = $("#rating").data("userrating");
+    review.rating = rating;
+  },
+  "click #submit": () => {
+    review.comment = document.getElementById("comment").value;
+    if (review.comment === "") {
+      Alerts.toast("Comment cannot be empty", "error");
+      return false;
```
Comment (emmyakin) [inline imports/plugins/included/product-detail-simple/client/templates/productReview.js]: remove return false

File: imports/plugins/included/product-detail-simple/client/templates/productReview.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const review = {};
+Template.submitReview.events({
+  "click .stars": () => {
+    const rating = $("#rating").data("userrating");
+    review.rating = rating;
+  },
+  "click #submit": () => {
+    review.comment = document.getElementById("comment").value;
+    if (review.comment === "") {
+      Alerts.toast("Comment cannot be empty", "error");
```
Comment (emmyakin) [inline imports/plugins/included/product-detail-simple/client/templates/productReview.js]: add return here

File: imports/plugins/included/product-detail-simple/client/templates/productReview.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      document.getElementById("comment").value = "";
+    } catch (error) {
+      Alerts.toast("You need to sign in to post a review", "error");
+    }
+  }
+});
+
+Template.displayReviews.helpers({
+  reviews: () => {
+    this.productId = () => Reaction.Router.getParam("handle");
```
Comment (emmyakin) [inline imports/plugins/included/product-detail-simple/client/templates/productReview.js:38]: why is this a function?

File: imports/plugins/included/product-detail-simple/client/templates/embedSocial.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      const baseUrl = "https://www.facebook.com/plugins/page.php?";
+      const href = `href=${facebookConfig.profilePage}`;
+      const dimensions = "&tabs=timeline&width=400&height=400&";
+      const sHeader = "small_header=false&adapt_container_width=true";
+      const remainder = "&hide_cover=false&show_facepile=false&";
+      const appID = `appId=${facebookConfig.appId}`;
+      url = `${baseUrl}${href}${dimensions}${sHeader}${remainder}${appID}`;
+      return url;
+    }
+    return false;
```
Comment (emmyakin) [inline imports/plugins/included/product-detail-simple/client/templates/embedSocial.js]: remove this

File: imports/plugins/included/product-detail-simple/client/templates/embedSocial.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  });
+});
+
+Template.embedTwitter.helpers({
+  twitter() {
+    const twitterConfig = Template.instance().state.get("feed").twitter;
+    if (twitterConfig.enabled && twitterConfig.profilePage) {
+      return twitterConfig.profilePage;
+    }
+    return false;
```
Comment (emmyakin) [inline imports/plugins/included/product-detail-simple/client/templates/embedSocial.js]: remove

**Review by emmyakin — CHANGES_REQUESTED** [review]
minor comments.

File: imports/plugins/included/product-detail-simple/client/templates/productReview.js (hunk 3 above)
Comment (mazma1) [reply imports/plugins/included/product-detail-simple/client/templates/productReview.js:38]: A function is required here to get the actual reference to the clicked product being reviewed. Using a regular variable will get the id of the product's ancestor (if the product has variants) which is not always the product being reviewed

**Review by emmyakin — APPROVED** [review]

**Pull request merged**